.buildkite/longruns/pipeline.yml

```diff
@@ -49,12 +54,59 @@ steps:
 
     steps:
 
-      - label: "AMIP"        
-        key: "mpi_amip_fine_long"
-        command: "mpiexec julia --color=yes --project=experiments/AMIP/moist_mpi_earth/ experiments/AMIP/moist_mpi_earth/coupler_driver.jl  --coupled true --surface_scheme monin_obukhov --moist equil --vert_diff true --rad clearsky --microphy 0M --z_elem 50 --dz_top 3000 --dz_bottom 30 --h_elem 16 --kappa_4 1e16 --rayleigh_sponge true --alpha_rayleigh_uh 0 --alpha_rayleigh_w 10 --dt_cpl 150 --dt 150 --dt_rad 6hours --idealized_insolation true --FLOAT_TYPE Float64 --energy_check false --mode_name amip --t_end 100days --dt_save_to_sol 10days --mono_surface false --run_name current_amip_n32_longrun"
-        artifact_paths: "experiments/AMIP/moist_mpi_earth/output/amip/current_amip_n32_longrun_artifacts/*"
+      - label: "Moist earth with slab surface - notmono + modular: bulk gray no_sponge idealinsol freq_dt_cpl"
+        key: "slabplanet"
+        command: "julia --color=yes --project=experiments/AMIP/moist_mpi_earth/ experiments/AMIP/moist_mpi_earth/coupler_driver_modular.jl --run_name default_modular_long --enable_threading true --coupled true --surface_scheme bulk  --moist equil --vert_diff true --rad gray --energy_check true --anim true --mode_name slabplanet --t_end 10days --dt_save_to_sol 1days --dt_cpl 200 --dt 200secs --mono_surface false --h_elem 4 --precip_model 0M"
+        artifact_paths: "experiments/AMIP/moist_mpi_earth/output/slabplanet/default_modular_long_artifacts/total_energy*.png"
+        env:
+          BUILD_HISTORY_HANDLE: ""
+        agents:
+          slurm_ntasks: 1
+          slurm_mem_per_cpu: 16G
+
+      - label: "AMIP"
+        key: "amip_longrun"
+        command: "mpiexec julia --color=yes --project=experiments/AMIP/moist_mpi_earth/ experiments/AMIP/moist_mpi_earth/coupler_driver_modular.jl --run_name target_amip_n64_longrun --enable_threading true --coupled true --surface_scheme monin_obukhov --moist equil --vert_diff true --rad gray --z_elem 50 --dz_top 3000 --dz_bottom 30 --h_elem 16 --kappa_4 1e16 --z_stretch false --rayleigh_sponge true --alpha_rayleigh_uh 0 --alpha_rayleigh_w 10 --dt_cpl 150 --dt 150secs --dt_rad 1hours --FLOAT_TYPE Float64 --energy_check false --mode_name amip --t_end 180days --dt_save_to_sol 10days --mono_surface false --precip_model 0M"
+        artifact_paths: "experiments/AMIP/moist_mpi_earth/output/amip/target_amip_n64_longrun_artifacts/*"
         env:
           CLIMACORE_DISTRIBUTED: "MPI"
+          BUILD_HISTORY_HANDLE: ""
+        agents:
+          slurm_ntasks: 64
+          slurm_mem_per_cpu: 16G
+
+      # Arguments passed from the ci pipeline.yml
+      - label: ":rocket: performance: flame graph diff: perf_target_amip_n1_shortrun"
+        command: "julia --color=yes --project=perf perf/flame_diff.jl --run_name 4"
+        artifact_paths: "perf/output/perf_diff_target_amip_n1_shortrun/*"
         agents:
-          slurm_ntasks: 32
+          slurm_mem: 20GB
```

Comment:
No, it's a fair question. In this case though, the performance benchmark runs only on one process (and one thread), so `slurm_tasks_per_node = slurm_ntasks = 1`. In the case above (where ntasks=64), we could minimize the number of nodes to 2, but this may mean delays until the scheduler finds nodes that are completely free, so it's a trade off. At the moment we're using this long MPI run for physics results, so I'd lean towards leaving it as is, so we can get the job done faster.

Reply:
That makes sense, thank you!